business/workloads.go

```diff
@@ -75,19 +75,96 @@ func isWorkloadIncluded(workload string) bool {
 }
 
 // GetWorkloadList is the API handler to fetch the list of workloads in a given namespace.
-func (in *WorkloadService) GetWorkloadList(namespace string) (models.WorkloadList, error) {
+func (in *WorkloadService) GetWorkloadList(namespace string, linkIstioResources bool) (models.WorkloadList, error) {
 	workloadList := &models.WorkloadList{
 		Namespace: models.Namespace{Name: namespace, CreationTimestamp: time.Time{}},
 		Workloads: []models.WorkloadListItem{},
 	}
-	ws, err := fetchWorkloads(in.businessLayer, namespace, "")
-	if err != nil {
+	var ws models.Workloads
+	var err error
+
+	var gateways []kubernetes.IstioObject
+	var authorizationPolicies []kubernetes.IstioObject
+	var peerAuthentications []kubernetes.IstioObject
+	var sidecars []kubernetes.IstioObject
+	var requestAuthentications []kubernetes.IstioObject
+	var envoyFilters []kubernetes.IstioObject
+
+	nFetches := 1
+	if linkIstioResources {
+		nFetches = 7
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(nFetches)
+	errChan := make(chan error, nFetches)
+
+	go func() {
+		defer wg.Done()
+		var err2 error
+		ws, err2 = fetchWorkloads(in.businessLayer, namespace, "")
+		if err2 != nil {
+			log.Errorf("Error fetching Workloads per namespace %s: %s", namespace, err2)
+			errChan <- err2
+		}
+	}()
+
+	linkedResources := map[string]*[]kubernetes.IstioObject{
+		kubernetes.Gateways:               &gateways,
+		kubernetes.AuthorizationPolicies:  &authorizationPolicies,
+		kubernetes.PeerAuthentications:    &peerAuthentications,
+		kubernetes.Sidecars:               &sidecars,
+		kubernetes.RequestAuthentications: &requestAuthentications,
+		kubernetes.EnvoyFilters:           &envoyFilters,
+	}
+
+	if linkIstioResources {
+		for linkedResource, dest := range linkedResources {
+			go func(namespace, resourceType string, dest *[]kubernetes.IstioObject, errChan chan error) {
+				defer wg.Done()
+				var err2 error
+				if IsNamespaceCached(namespace) {
+					*dest, err2 = kialiCache.GetIstioObjects(namespace, resourceType, "")
+				} else {
+					*dest, err2 = in.k8s.GetIstioObjects(namespace, resourceType, "")
+				}
+				if err2 != nil {
+					log.Errorf("Error fetching Istio %s per namespace %s: %s", resourceType, namespace, err2)
+					errChan <- err2
+				}
+			}(namespace, linkedResource, dest, errChan)
+		}
+	}
```

Comment:
I guess you missed it here as well. It can be refactored to avoid static declarations for each type.